src/TemplateProcessor.ts

```diff
@@ -1090,6 +1099,64 @@ export default class TemplateProcessor {
         return null;
     }
 
+
+    async waitCondition(waitCondition, timeout = 10000) {
+        return new Promise((resolve, reject) => {
+            const startTime = Date.now();
+            let conditionExpression;
+            let timeoutId;  // Declare timeoutId here
+
+            try {
+                conditionExpression = jsonata(waitCondition);
+            } catch (e) {
+                this.logger.error(`invalid wait condition expression: ${waitCondition}, ${e}`);
```

Comment:
I am not sure this is the right approach. I try to wrap functions in safe() so that I can simply throw error and rely on the framework to catch it and pack it into an {error:{...}} object

Reply:
I've tried a few different approaches. I like with withErrorHandling to handling calling to an external function, where we don't know the context. In this case, I am handling errors from my own code execution, and I think we can do better with a custom error messages.

Reply:
but your custom error message *will* be used by withErrorHandling. All you need to do is throw an error:
```
private withErrorHandling(fn) {
        return (...args) => {
            try {
                const result = fn(...args);
                if (result instanceof Promise) {
                    return result.catch(error => {
                        this.logger.error(error.toString());
                        return {
                            "error": {
                                message: error.message,
                                name: error.name,
                                stack: error.stack,
                            }
                        };
                    });
                }
                return result;
            } catch (error) {
                this.logger.error(error.toString());
                return {
                    "error": {
                        message: error.message,
                        name: error.name,
                        stack: error.stack,
                    }
                };
            }
        };
    };

Reply:
^^^ see how withErrorHandling preserves these properties. So why don't you just catch(e) and set it's message, and rethrow it?